### ISSUE TITLE
Should the main module on this repo be made internal?

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector-releases#84]
On the v0.47.0 release we had an issue when releasing binaries (see open-telemetry/opentelemetry-collector/issues/5013) that was not caught on CI until after tagging `v0.47.0`. Having a public Go module on this repository makes it hard to untag and re-tag a release without potentially breaking people's builds.

If we don't intend the main module on this repository to be publicly used (or, at least, if we don't intend it for now), we should place it in an `internal` folder so that we can safely re-tag if we have the need.

cc @jpkrohling to evaluate this

**Comment by jpkrohling** [comment]
It should indeed be internal. 

@silverlyra, what do you think?

**Comment by Aneurysm9** [comment]
I think as @jpkrohling mentioned [here](https://github.com/open-telemetry/opentelemetry-collector-releases/pull/83/files#r829097273), this should be handled through configuration and not code changes.  Even if it does require a code change, I'm not sure that mutable tags are ever the correct answer.  We've seen enough problems caused by others doing that, can we please not become part of the problem by normalizing that behavior?

**Comment by jpkrohling** [comment]
I agree in principle, but this specific repo is only used for releases. The only deliverables are the containers and binaries, and as long as they are not published, it's safe to remove and retag.

That said, we should only do it after careful consideration and make a decision as a team. The default answer should indeed be to produce a patch release.

I still think retagging v0.47.0 on this repository here was the right answer.

**Comment by mx-psi** [comment]
I second @jpkrohling with this. I would never suggest doing this on `opentelemetry-collector` or `opentelemetry-collector-contrib`, but this repo is only used to publish releases. 

I think we should avoid doing this as much as possible but, in the event that this is the best option in the future, making the module `internal` will help avoid breakage, and it is also the right thing to do generally if we don't want people to rely on it.

I am interested in hearing what you think would break from this on this repository, maybe we are missing something important

**Comment by mx-psi** [comment]
I tried doing this on #172. There is no public API, because if a package has a `main` function, it can't be imported. Therefore this is not a problem for moving tags in special situations.
